source/Octopus.Tentacle/Configuration/TentacleConfiguration.cs

```diff
@@ -55,7 +59,7 @@ internal class TentacleConfiguration : ITentacleConfiguration
         [Obsolete("This configuration entry is obsolete as of 3.0. It is only used as a Subscription ID where one does not exist.")]
         public string? TentacleSquid => settings.Get("Octopus.Communications.Squid", (string?)null);
 
-        public IEnumerable<OctopusServerConfiguration> TrustedOctopusServers => settings.Get<IEnumerable<OctopusServerConfiguration>>(TrustedServersSettingName) ?? new OctopusServerConfiguration[0];
+        public IEnumerable<OctopusServerConfiguration> TrustedOctopusServers => changeDetectingSettings.Get<IEnumerable<OctopusServerConfiguration>>(TrustedServersSettingName) ?? Array.Empty<OctopusServerConfiguration>();
```

Comment:
`TrustedOctopusServers` will detect changes to the tentacle configuration so Trust changes do not require a restart of Tentacle